pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -89,6 +90,21 @@ func (m *kubeGenericRuntimeManager) generateLinuxContainerConfig(container *v1.C
 
 	lc.Resources.HugepageLimits = GetHugepageLimitsFromResources(container.Resources)
 
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeSwapEnabled) {
+		// NOTE(ehashman): Behaviour is defined in the opencontainers runtime spec:
+		// https://github.com/opencontainers/runtime-spec/blob/1c3f411f041711bbeecf35ff7e93461ea6789220/config-linux.md#memory
+		switch m.memorySwapBehavior {
+		case kubelettypes.UnlimitedSwap:
+			// -1 = unlimited swap
+			lc.Resources.MemorySwapLimitInBytes = -1
```

Comment:
just recording this as this swaps out of my own mental cache sometimes.

if a container has a defined memory limit X, it will still have `MemoryLimitInBytes=X`, but it may now use unbounded additional swap by setting `MemorySwapLimitInBytes` if `UnlimitedSwap` is enabled.  This is consistent with existing behavior where --fail-swap-on was false because no kubelet enforced limit was written.